tests/test_events.py

```diff
@@ -224,16 +228,32 @@ def test_process_block_range(client, base_ddo_url, events_object):
         assert events_object.process_current_blocks() is None
 
 
+def test_elasticsearch_connection(events_object, caplog):
+    with patch("elasticsearch.Elasticsearch.ping") as es_mock:
+        es_mock.return_value = True
+        with patch("elasticsearch.Elasticsearch.get") as mock:
+            mock.return_value = {"_source": {"last_block": 24}}
+            assert events_object.get_last_processed_block() == 24
+
+    with patch("elasticsearch.Elasticsearch.ping") as es_mock:
+        es_mock.return_value = False
+        action_thread = threading.Thread(target=events_object.get_last_processed_block)
+        action_thread.start()
+        time.sleep(5)
+        es_mock.return_value = True
+        action_thread.join()
+        assert "Connection to ES failed. Trying to connect to back..." in caplog.text
+
+
 def test_get_last_processed_block(events_object):
-    start_block = events_object._start_block
     with patch("elasticsearch.Elasticsearch.get") as mock:
         mock.side_effect = Exception("Boom!")
-        assert events_object.get_last_processed_block() == start_block
+        assert events_object.get_last_processed_block() == 0
```

Comment:
well, not really. We can have a custom start block which is not 0. The only reason it's 0 for ganache it's because we define it to be. Look into the list of env vars and you'll find what I'm talking about.

Reply:
I will assert with BFACTORY_BLOCK, right?